Importing Library

In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

Loading the dataset and storing it in the df

In [3]:
df = pd.read_csv('Weather_Data.csv')

In [4]:
df.head()

STATION                 NAME        DATE  PRCP  TAVG  TMAX  TMIN
0  IN022023000  NEW DELHI PALAM, IN  10-10-2022  0.00    72  76.0  69.0
1  IN022023000  NEW DELHI PALAM, IN  11-10-2022  0.01    77   NaN  70.0
2  IN022023000  NEW DELHI PALAM, IN  12-10-2022  0.00    77   NaN   NaN
3  IN022023000  NEW DELHI PALAM, IN  13-10-2022   NaN    77  86.0   NaN
4  IN022023000  NEW DELHI PALAM, IN  14-10-2022   NaN    78  87.0  69.0

Creating the Date as the index

In [5]:
df = pd.read_csv('Weather_Data.csv',index_col='DATE')

In [6]:
df.head()

STATION                 NAME  PRCP  TAVG  TMAX  TMIN
DATE                                                                
10-10-2022  IN022023000  NEW DELHI PALAM, IN  0.00    72  76.0  69.0
11-10-2022  IN022023000  NEW DELHI PALAM, IN  0.01    77   NaN  70.0
12-10-2022  IN022023000  NEW DELHI PALAM, IN  0.00    77   NaN   NaN
13-10-2022  IN022023000  NEW DELHI PALAM, IN   NaN    77  86.0   NaN
14-10-2022  IN022023000  NEW DELHI PALAM, IN   NaN    78  87.0  69.0

Infomation of the columns of the dataset
Date->Date of the record
Station->Name of the station
Name->Name of the place
PRCP->Precipation
TAVG->Represent average hourly values
TMAX->Represent average hourly maximum values
TMIN->Represent average hourly minimum values

Checking the shape of the dataset

In [9]:
df.shape

(750, 6)

There are total 750 rows and 6 columns

Checking the duplicates in the dataset

In [8]:
df.duplicated().sum()

40

Dropping the duplicated rows in the dataset

In [19]:
df = df.drop_duplicates()

In [20]:
df.head()

STATION                 NAME  PRCP  TAVG  TMAX  TMIN
DATE                                                                
10-10-2022  IN022023000  NEW DELHI PALAM, IN  0.00    72  76.0  69.0
11-10-2022  IN022023000  NEW DELHI PALAM, IN  0.01    77   NaN  70.0
12-10-2022  IN022023000  NEW DELHI PALAM, IN  0.00    77   NaN   NaN
13-10-2022  IN022023000  NEW DELHI PALAM, IN   NaN    77  86.0   NaN
14-10-2022  IN022023000  NEW DELHI PALAM, IN   NaN    78  87.0  69.0

Checking the null values in the dataset

In [21]:
df.isnull().sum()

STATION      0
NAME         0
PRCP       512
TAVG         0
TMAX       148
TMIN       157
dtype: int64

In [22]:
df.isnull().sum().sum()

817

There are total 817 null values in the dataset

In [33]:
# Creating the function for replacing the null values
def replace_null(df,columns) :
    df.fillna(value=df[columns].median(),inplace=True)
    return df

In [34]:
# Replacing all the null values with the help of the user defined function
replace_null(df,'PRCP')
replace_null(df,'TMAX')
replace_null(df,'TMIN')

STATION                     NAME  PRCP  TAVG   TMAX   TMIN
DATE                                                                      
10-10-2022  IN022023000      NEW DELHI PALAM, IN  0.00    72  76.00  69.00
11-10-2022  IN022023000      NEW DELHI PALAM, IN  0.01    77   0.02  70.00
12-10-2022  IN022023000      NEW DELHI PALAM, IN  0.00    77   0.02   0.02
13-10-2022  IN022023000      NEW DELHI PALAM, IN  0.02    77  86.00   0.02
14-10-2022  IN022023000      NEW DELHI PALAM, IN  0.02    78  87.00  69.00
...                 ...                      ...   ...   ...    ...    ...
15-10-2023  IN022021900  NEW DELHI SAFDARJUN, IN  0.02    81  98.00  71.00
16-10-2023  IN022021900  NEW DELHI SAFDARJUN, IN  0.02    76  94.00   0.02
17-10-2023  IN022021900  NEW DELHI SAFDARJUN, IN  0.20    72  87.00  63.00
18-10-2023  IN022021900  NEW DELHI SAFDARJUN, IN  0.00    73   0.02  63.00
19-10-2023  IN022021900  NEW DELHI SAFDARJUN, IN  0.02    73  87.00  61.00

[710 rows x 6 columns]